File: sdks/ruby/lib/novabilling/types/top_up_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class TopUpResponse < Internal::Types::Model
      field :transactions, -> { Internal::Types::Array[Novabilling::Types::WalletTransactionResponse] }, optional: false, nullable: false
      field :wallet, -> { Novabilling::Types::WalletResponse }, optional: false, nullable: false
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/TopUpResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record TopUpResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("transactions")]
    public IEnumerable<WalletTransactionResponse> Transactions { get; set; } =
        new List<WalletTransactionResponse>();

    [JsonPropertyName("wallet")]
    public required WalletResponse Wallet { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
